crates/bitcoin-da/tests/bitcoin_da_tests.rs

```diff
@@ -323,61 +341,96 @@ mod tests {
     fn test_reveal2() {
+        // ======================================
+        // Given: a configured Bitcoin relayer on the REGNET network with embedded data
+        // ======================================
+    
+        // Set up embedded data and relayer configuration
         let embedded_data = b"Hello, world!";
         let relayer = Relayer::new(&Config::new(
-            // "YOUR_NODE_IP:38332".to_owned(), // SIGNET
-            "localhost:8332".to_owned(), // REGNET
+            NODE_IP.to_owned(),
             "rpcuser".to_owned(),
             "rpcpass".to_owned(),
         ))
         .unwrap();
-        // get network, should be regtest
+    
+        // Validate the network is indeed Regtest (setup validation)
         let blockchain_info = relayer.client.get_blockchain_info().unwrap();
         let network_name = &blockchain_info.chain;
         let network = Network::from_core_arg(network_name)
             .map_err(|_| BitcoinError::InvalidNetwork)
             .unwrap();
         assert_eq!(network, Network::Regtest);
-        // append id to data
+    
+        // Append protocol id to data
         let mut data_with_id = Vec::from(&PROTOCOL_ID[..]);
         data_with_id.extend_from_slice(embedded_data);
-        // create address with data in script
+    
+        // Create a taproot address with the data embedded in the script
         let address = create_taproot_address(&data_with_id, network).unwrap();
-        // do first transaction -> commit
-        match relayer.commit_tx(&address) {
-            Ok(txid) => match relayer.reveal_tx(&data_with_id, &txid) {
-                Ok(txid) => {
-                    println!("Reveal Txid: {}", txid);
-                    println!("Successful Reveal");
+    
+        // ================================================
+        // When: a commit transaction is made, followed by a reveal transaction
+        // ================================================
+    
+        let commit_result = relayer.commit_tx(&address);
+    
+        // ===============================================
+        // Then: assert the success of the reveal operation after a successful commit
+        // ===============================================
+    
+        match commit_result {
+            Ok(txid) => {
+                match relayer.reveal_tx(&data_with_id, &txid) {
+                    Ok(txid) => {
+                        println!("Reveal Txid: {}", txid);
+                        println!("Successful Reveal");
+                    }
+                    Err(e) => panic!("Reveal failed with error: {:?}", e),
                 }
-                Err(e) => panic!("Reveal failed with error: {:?}", e),
-            },
+            }
             Err(e) => panic!("Commit failed with error: {:?}", e),
         }
     }
+    
 
     #[test]
     fn test_write() {
+        // ===============================
+        // Given: a configured Bitcoin relayer on the REGNET network
```

Comment:
Would be better to keep both ways signet/regtest. 
At least for us, it is important to be able to switch fast aka commenting/uncommenting.
Ugly but efficient

Reply:
Nop, looks good like that !